### PR TITLE
chore: copy over explorer v0.18.4

### PR DESCRIPTION
**Pull request by ra0x3**
### Description
- Copies over the block explorer from the indexer repo v0.18.4
- We will continue to migrate changes over, as we change the explorer
- However, there has to probably be a better way than copying files over?
  - Copying isn't terrible, it's just you have to make a few changes to the files after you copy them over (since the original files come from a different workspace) 

### Testing steps
- [ ] cd `fuel-explorer/packages/fuel-explorer`
- [ ] `forc index build` (using `forc index v0.18.4`)

### CODE REVIEW

**Review by Braqzen — COMMENTED** [review]
In the README I'm stuck on spinning up the container. I've left it running for a while and it's stuck on attaching `Attaching to fuel-explorer-fuel-indexer-1, fuel-explorer-postgres-1`.

Deploying the indexer also fails as it doesn't find some directory.

@luizstacio may be able to explain the remote merge to keep history rather than copy / pasting

File: packages/fuel-explorer/fuel-explorer/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,925 @@
...
+            id: id8(block_data.header.id),
+            block_id: block_data.header.id,
+            header: header.id,
+            consensus: consensus.id,
+            transactions: Some(tx_id_frags),
+        };
+
+        block.save();
+
+        let _transactions = block_data
```
[the remark this answers is not in the document]
Comment (ra0x3) [reply packages/fuel-explorer/fuel-explorer/src/lib.rs:919]: @Braqzen correct

File: packages/fuel-explorer/fuel-explorer/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,925 @@
...
+            block_data.header.transactions_root,
+            block_data.header.output_messages_root,
+            block_data.header.height,
+            block_data.header.prev_root,
+            block_data.header.time,
+            block_data.header.application_hash,
+        );
+        header.save();
+
+        let _block_frag = BlockIdFragment::from(block_data.header.id);
```
[the remark this answers is not in the document]
Comment (ra0x3) [reply packages/fuel-explorer/fuel-explorer/src/lib.rs:897]: [See comment](https://github.com/FuelLabs/fuel-explorer/pull/25#discussion_r1279339516)

File: packages/fuel-explorer/fuel-explorer/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,925 @@
+extern crate alloc;
+use fuel_indexer_utils::prelude::*;
+
+// TODO: We use a lot of manual type conversion below due to https://github.com/FuelLabs/fuel-indexer/issues/286
```
[the remark this answers is not in the document]
Comment (Braqzen) [reply packages/fuel-explorer/fuel-explorer/src/lib.rs:4]: It is a preference for UX and I didn't say that it wasn't straightforward. 
My point of having to scroll to find what you're looking for has not been countered but okay.

**Review by Braqzen — APPROVED** [review]
utACK for the deployment

**Review by luizstacio — CHANGES_REQUESTED** [review]
We need to keep the basic infra, I saw on this PR we added a new docker-compose, and removed basic infra package.json and scripts.

The current docker on the root folder already handles spinning a fuel-core with data inside.

Do you think we should change this? Why?

**Comment by ra0x3** [comment]
@luizstacio Doh! Didn't intend to remove anything, but looks like some files were left off when I renamed the package. I'll add those back 👌🏽

**Pull request closed without merging**